internal/ingress/controller/controller_test.go

```diff
@@ -268,6 +268,23 @@ func TestCheckIngress(t *testing.T) {
 			}
 		})
 
+		t.Run("When invalid directives are used in annotation values", func(t *testing.T) {
+			nginx.store = fakeIngressStore{
+				ingresses: []*ingress.Ingress{},
+				configuration: ngx_config.Configuration{
+					AnnotationValueWordBlocklist: "invalid_directive, another_directive",
+				},
+			}
+			nginx.command = testNginxTestCommand{
+				t:   t,
+				err: nil,
+			}
+			ing.ObjectMeta.Annotations["nginx.ingress.kubernetes.io/custom-headers"] = "invalid_directive"
```

Comment:
With context of my previous comment, checking for `another_directive` would also perhaps be useful.

Reply:
https://github.com/kubernetes/ingress-nginx/pull/7921/files

Added here just to make sure it's working fine :)